respect-lib-shared/src/commonMain/kotlin/world/respect/shared/viewmodel/report/edit/ReportEditViewModel.kt

```diff
@@ -0,0 +1,413 @@
+package world.respect.shared.viewmodel.report.edit
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.toRoute
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.serialization.json.Json
+import org.jetbrains.compose.resources.getString
+import world.respect.datalayer.DataReadyState
+import world.respect.datalayer.respect.RespectReportDataSource
+import world.respect.datalayer.respect.model.Indicator
+import world.respect.datalayer.respect.model.RespectReport
+import world.respect.shared.domain.report.model.DefaultIndicators
+import world.respect.shared.domain.report.model.RelativeRangeReportPeriod
+import world.respect.shared.domain.report.model.ReportFilter
+import world.respect.shared.domain.report.model.ReportOptions
+import world.respect.shared.domain.report.model.ReportSeries
+import world.respect.shared.domain.report.model.ReportSeriesVisualType
+import world.respect.shared.ext.replace
+import world.respect.shared.generated.resources.Res
+import world.respect.shared.generated.resources.add_a_new_report
+import world.respect.shared.generated.resources.done
+import world.respect.shared.generated.resources.edit_report
+import world.respect.shared.generated.resources.field_required_prompt
+import world.respect.shared.generated.resources.quantity_must_be_at_least_1
+import world.respect.shared.generated.resources.series
+import world.respect.shared.navigation.IndicatorList
+import world.respect.shared.navigation.NavCommand
+import world.respect.shared.navigation.NavResultReturner
+import world.respect.shared.navigation.ReportDetail
+import world.respect.shared.navigation.ReportEdit
+import world.respect.shared.navigation.ReportEditFilter
+import world.respect.shared.resources.StringResourceUiText
+import world.respect.shared.resources.UiText
+import world.respect.shared.util.ext.asUiText
+import world.respect.shared.viewmodel.RespectViewModel
+import world.respect.shared.viewmodel.app.appstate.ActionBarButtonUiState
+import world.respect.shared.viewmodel.app.appstate.AppUiState
+import world.respect.shared.viewmodel.app.appstate.LoadingUiState
+
+data class ReportEditUiState(
+    val reportOptions: ReportOptions = ReportOptions(),
+    val reportTitleError: UiText? = null,
+    val xAxisError: UiText? = null,
+    val seriesTitleErrors: Map<Int, UiText> = emptyMap(),
+    val yAxisErrors: Map<Int, UiText> = emptyMap(),
+    val subGroupError: UiText? = null,
+    val chartTypeError: Map<Int, UiText> = emptyMap(),
+    val timeRangeError: UiText? = null,
+    val quantityError: UiText? = null,
+    val submitted: Boolean = false,
+    val hasSingleSeries: Boolean = true,
```

Comment:
hasSingleSeries should not exist: this is not respecting single point of truth and creates the potential for an inconsistent state. Remove it.

What you can do:

e.g.
```
val hasSingleSeries: Boolean
    get() = reportOptions.series.size == 1
```